### ISSUE TITLE
Solidify loss function invariant; determine where to convert labels to matrices

### ISSUE THREAD

**Issue by fufjvnvnf** [fufjvnvnf/NN.jl#3]

**Comment by fufjvnvnf** [comment]
Solved by leaving old `SoftMaxCrossEntropy` alone and not output `prediction` in the new loss function layer.
